ambari-agent/src/main/python/ambari_agent/CustomServiceOrchestrator.py

```diff
@@ -456,6 +459,43 @@ def runCommand(self, command_header, tmpoutfile, tmperrfile, forced_command_name
       if incremented_commands_for_component:
         self.commands_for_component_in_progress[cluster_id][command['role']] -= 1
 
+      # Conditionally remove the command-*.json file if it exists
+      if os.path.exists(json_path):
+        command_file_retention_policy = self.config.command_file_retention_policy
+
+        if command_file_retention_policy == self.config.COMMAND_FILE_RETENTION_POLICY_REMOVE:
+          remove_command_file = True
+          logger.info(
+            'Removing %s due to the command_file_retention_policy, %s',
+            json_path, command_file_retention_policy
+          )
+        elif command_file_retention_policy == self.config.COMMAND_FILE_RETENTION_POLICY_REMOVE_ON_SUCCESS:
+          if ret and ('exitcode' in ret):
+            exit_code = ret['exitcode']
+            if exit_code == 0:
+              remove_command_file = True
+              logger.info(
+                'Removing %s due to the command_file_retention_policy, %s, and exit code, %d',
+                json_path, command_file_retention_policy, exit_code
+              )
+            else:
+              remove_command_file = False
+              logger.info(
+                'Not removing %s due to the command_file_retention_policy, %s, and exit code, %d',
+                json_path, command_file_retention_policy, exit_code
```

Comment:
I don't think we need to log this behavior.

Reply:
really.. I was thinking that the was the only logging we really needed since it was an unexpected or undesired scenario.

Reply:
How about I change the logging to be `debug` rather than `info`?

Reply:
These logs would be very helpful when we need to figure out if the command JSON became removed or did not arrived...+1 for logging

Reply:
I suppose that's fine if you think it's information that could be needed. On a normal basis, I would say it's not, but we can leave it as debug.

Reply:
Oh, did I make a mistake where I put my comment - I only wanted the normal "keep" behavior logging removed.

Reply:
There is no logging for the normal keep path.